cmd/packer-plugin-check/main.go

```diff
@@ -0,0 +1,178 @@
+// packer-plugin-check is a command used by plugins to validate compatibility and basic configuration
+// to work with Packer.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/hashicorp/packer/packer/plugin"
+)
+
+const packerPluginCheck = "packer-plugin-check"
+
+var (
+	hcl2spec = flag.Bool("hcl2spec", false, "flag to indicate that hcl2spec files should be checked.")
+	docs     = flag.Bool("docs", false, "flag to indicate that documentation files should be checked.")
+	load     = flag.String("load", "", "flag to check if plugin can be loaded by Packer.")
+)
+
+// Usage is a replacement usage function for the flags package.
+func Usage() {
+	fmt.Fprintf(os.Stderr, "Usage of "+packerPluginCheck+":\n")
+	fmt.Fprintf(os.Stderr, "\t"+packerPluginCheck+" [flags]\n")
+	fmt.Fprintf(os.Stderr, "Flags:\n")
+	flag.PrintDefaults()
+}
+
+func main() {
+	log.SetFlags(0)
+	log.SetPrefix(packerPluginCheck + ": ")
+	flag.Usage = Usage
+	flag.Parse()
+
+	if *hcl2spec == false && *docs == false && len(*load) == 0 {
+		flag.Usage()
+		os.Exit(2)
+	}
+
+	if *hcl2spec {
+		if err := checkHCL2Specs(); err != nil {
+			fmt.Printf(err.Error())
+			os.Exit(2)
+		}
+		fmt.Printf("Plugin succesfully passed hcl2spec check.\n")
+	}
+
+	if *docs {
+		if err := checkDocumentation(); err != nil {
+			fmt.Printf(err.Error())
+			os.Exit(2)
+		}
+		fmt.Printf("Plugin succesfully passed docs check.\n")
+	}
+
+	if len(*load) > 0 {
+		if err := checkPluginName(*load); err != nil {
+			fmt.Printf(err.Error())
+			os.Exit(2)
+		}
+		if err := discoverAndLoad(); err != nil {
+			fmt.Printf(err.Error())
+			os.Exit(2)
+		}
+		fmt.Printf("Plugin succesfully passed compatibility check.\n")
+	}
+}
+
+// checkHCL2Specs looks for the presence of a hcl2spec.go file in the current directory.
+// It is not possible to predict the number of hcl2spec.go files for a given plugin.
+// Because of that, finding one file is enough to validate the knowledge of hcl2spec generation.
+func checkHCL2Specs() error {
+	wd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	var hcl2found bool
+	_ = filepath.Walk(wd, func(path string, info os.FileInfo, err error) error {
+		if info.IsDir() {
+			if info.Name() == "docs" || info.Name() == ".github" {
+				return filepath.SkipDir
+			}
+		} else {
+			if strings.HasSuffix(path, "hcl2spec.go") {
+				hcl2found = true
+				return io.EOF
+			}
+		}
+		return nil
+	})
+
+	if hcl2found {
+		return nil
+	}
```

Comment:
nit: Since the if IsDir() is a guard clause you can drop the else statement

```go
	_ = filepath.Walk(wd, func(path string, info os.FileInfo, err error) error {
		if info.IsDir() {
			if info.Name() == "docs" || info.Name() == ".github" {
				return filepath.SkipDir
			}
			return nil
		} 
		
		if strings.HasSuffix(path, "hcl2spec.go") {
			hcl2found = true
			return io.EOF
		}
		return nil
	})
```

Reply:
good catch, thx!